packages/service-utils/src/cf-worker/usageV2.ts

```diff
@@ -34,19 +28,26 @@ export async function sendUsageV2Events<T extends UsageV2Options>(
     : ClientUsageV2Event[],
   options: T,
 ): Promise<void> {
+  const { usageBaseUrl, source, authInput, serviceKey } = options;
+  const { clientId, secretKey } = await extractAuthorizationData(authInput);
+
   // Determine endpoint and auth header based on provided credentials.
   let url: string;
   const headers: HeadersInit = { "Content-Type": "application/json" };
+  const origin = authInput.req.headers.get("Origin");
```

Comment:
Debating if I should just shallow copy all CF headers to the call to usage tracker.